sql/core/src/test/scala/org/apache/spark/sql/execution/SQLViewTestSuite.scala

```diff
@@ -293,22 +295,45 @@ abstract class SQLViewTestSuite extends QueryTest with SQLTestUtils {
       }
     }
   }
+
+  test("SPARK-34152: view's identifier should be correctly stored") {
+    Seq(true, false).foreach { storeAnalyzed =>
+      withSQLConf(STORE_ANALYZED_PLAN_FOR_VIEW.key -> storeAnalyzed.toString) {
```

Comment:
This has no effect for permanent view, and please let me know if this is not desirable. I can introduce `isTempView` for each suite and only apply this if the suite is for a local or global temp view.

Reply:
it's ok to have a little waste in the test.